src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -58,7 +59,7 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
             this.config = config;
             numMessages = 0;
 
-            logger = runtime.GetLogger(((ISystemTargetBase)this).GrainId + "-" + streamProviderName);
+            logger = runtime.ServiceProvider.GetRequiredService<ILoggerFactory>().CreateLogger($"{this.GetType().Namespace}.{((ISystemTargetBase)this).GrainId}.{streamProviderName}");
```

Comment:
Should this include the type name in addition to the namespace?

Reply:
I think this is a special case, since the old one doesn't use its name too. Adding namespace mainly to make logging filters work properly. Didn't use FullName mainly for the concern of category name too long. 

But I think you are right, we should use FullName, especially for the case if user apply log filters specific to this type's full name